cmd/otel-allocator/collector/collector.go

```diff
@@ -112,14 +111,14 @@ func runWatch(ctx context.Context, k *Client, c <-chan watch.Event, collectorMap
 			return "context done"
 		case event, ok := <-c:
 			if !ok {
```

Comment:
`!ok` means the channel is closed and empty so I think restarting the Watch when this happens is definitely what we want.  I saw below (L131) that we have wait for our timeout with `time.After(watcherTimeout)` but I'm pretty sure we can add a timeout `opt` on the `Watch()` call and then consider `!ok` to be the Watch ending / timing out.

Reply:
Thanks for the feedback! I tried taking this suggestion but for some reason `runWatch` just hangs when I try to add the timeout to the [ListOptions](https://github.com/open-telemetry/opentelemetry-operator/pull/1237/files#diff-9c164a07302845ef1a30b0d2893d22d8fb254f69f8e868e957f35c587baa462bR56), so my unit test just hangs.  Currently my unit test for timeout works only if I include this [block](https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/otel-allocator/collector/collector.go#L132) in runWatch.  Hmm I'm wondering if this timeout is only relevant in the specific case that `k8sClient.CoreV1().Pods(ns).List(ctx, opts)`  hangs, but I also just might be using this incorrectly in my test

Reply:
This is solved with the context timeout above. 👍